app/routes/synonyms.$synonymId.tsx

```diff
@@ -61,81 +98,103 @@ export async function action({
 }
 
 export default function () {
-  const { eventIds } = useLoaderData<typeof loader>()
+  const { eventIds, options } = useLoaderData<typeof loader>()
+  const uniqueOptions = Array.from(new Set(options)) as string[]
   const [deleteSynonyms, setDeleteSynonyms] = useState([] as string[])
   const [synonyms, setSynonyms] = useState(eventIds || [])
   const [addSynonyms, setAddSynonyms] = useState([] as string[])
```

Comment:
Why are all of these type assertions (`X as Y`) necessary? Why is it failing to infer types?

Reply:
That is because of a quirk in how react useState behaves. [Here is a video](https://www.totaltypescript.com/tips/avoid-unexpected-behaviour-of-react-s-usestate) that shows the exact case. If you pass useState an empty array, it infers that the type is a never[]. So if you try to add anything to it, you can't because it's a never[]. 
<img width="1089" alt="Screenshot 2024-10-10 at 10 27 56 AM" src="https://github.com/user-attachments/assets/7c6e7f90-586b-400c-8b4d-76f6dc83b580">
You can solve for the issue by using useState<string[]>([]) or by casting the array to a string array like I did here. I'll go ahead and change it to to use useState<string[]>([]) since you probably prefer the terseness and not casting.

Reply:
Yes, I prefer that. Better than overriding TypeScript. Thanks!